File: A_Pythagorean_Theorem_II.cpp
#include <bits/stdc++.h>
using namespace std;

int main(){
    int n;
    cin >> n;

    int counter = 0;
    for (int a = 1; a <= n; a++){
        for (int b = a; b <= n; b ++){
            int c2 = a * a + b * b;
            int c = sqrt(c2);

            if (c2 == c * c and c <= n){
                counter++;
            }
        }
    }

    cout << counter << "\n";
}

File: A_Pythagorean_Theorem_II.py
# rating: 1200

import math

n = int(input())

counter = 0
for a in range(1, n):
    for b in range(a + 1, n):
        c = math.sqrt(a * a + b * b)

        if int(c) == c and c <= n:
            counter += 1

print(counter)